### PR TITLE
add more tests

### PR DESCRIPTION
**Pull request by sakrist**
Added missing functions and few more tests

### CODE REVIEW

**Comment by sakrist** [comment]
@s1ddok here is few more tests and fixes

File: Sources/Matrix3x3+nosimd.swift [hunk 1]
```diff
@@ -148,12 +148,14 @@ public struct Matrix3x3f {
             - (m13 * m22 * m31 + m11 * m23 * m32 + m12 * m21 * m33)
     }
     
-    public var transpose: Matrix3x3f {
-        return Matrix3x3f(m11, m21, m31, m12, m22, m32, m13, m23, m33)
+    public var transposed: Matrix3x3f {
+        return Matrix3x3f(m11, m12, m13,
```
Comment (s1ddok) [inline Sources/Matrix3x3+nosimd.swift:152]: it is really uncomfortable for me every time to remember if we do column or row-major haha

Comment (s1ddok) [reply Sources/Matrix3x3+nosimd.swift:152]: okay it seems correct

**Comment by s1ddok** [comment]
thank you! you found so many bugs in the library!

File: Sources/Matrix3x3+nosimd.swift (hunk 1 above)
Comment (sakrist) [reply Sources/Matrix3x3+nosimd.swift:152]: yea, it is confusing here. I suggest to rewrite matrix3x3 and matrix4x4

**Pull request merged**